scylla/src/transport/cluster.rs

```diff
@@ -30,19 +29,11 @@ pub struct Cluster {
     _worker_handle: RemoteHandle<()>,
 }
 
-#[derive(Clone)]
-pub struct Datacenter {
-    pub nodes: Vec<Arc<Node>>,
-    pub rack_count: usize,
-}
-
 #[derive(Clone)]
 pub struct ClusterData {
     pub(crate) known_peers: HashMap<SocketAddr, Arc<Node>>, // Invariant: nonempty after Cluster::new()
     pub(crate) ring: BTreeMap<Token, Arc<Node>>, // Invariant: nonempty after Cluster::new()
     pub(crate) keyspaces: HashMap<String, Keyspace>,
-    pub(crate) all_nodes: Vec<Arc<Node>>,
-    pub(crate) datacenters: HashMap<String, Datacenter>,
```

Comment:
I don't think that the primary use case for those fields is valid anymore. They acted as an lookup tables to be used by load balancing policies. Now, with the arrival of the new, event based load balancing policy trait, policies can build this lookup tables themselves (by listening to topology events).

I'm all for removing `all_nodes` - it's just `known_peers.values().collect()` and I don't see a situation (other than old load balancing impl) where one needs to use integer indexing (provided by `Vec`) instead of indexing nodes by their address (`HashMap<SocketAddr, _>`).

`datacenters` seems to be more useful than `all_nodes`, but I don't see a use case either.

Reply:
OK. If it is possible to obtain the info those fields expose with other means (even if it is a bit slower) then I think it's alright.

Besides, I missed that it was `pub(crate)`, so it's not visible to users anyway...

Reply:
> Besides, I missed that it was pub(crate), so it's not visible to users anyway...

There are `pub` getters: `ClusterData::get_nodes_info` and `ClusterData::get_datacenters_info`.